dspace-api/src/test/java/org/dspace/content/logic/LogicalFilterTest.java

```diff
@@ -0,0 +1,358 @@
+            fail("LogicalStatementException thrown testing the NAND operator" + e.getMessage());
+        }
+    }
+
+    /**
+     * Test the NOR operator with simple lists of logical statements
+     */
+    @Test
+    public void testNorOperator() {
+        // Blank operator
+        Nor nor = new Nor();
+        // Try tests - the item can be null, as the statements are simply returning booleans themselves
+        try {
+            // Set to True, True (expect False)
+            nor.setStatements(trueStatements);
+            assertFalse("NOR operator did not return false for a list of true statements",
+                nor.getResult(context, null));
+            // Set to True, False (expect False)
+            nor.setStatements(trueFalseStatements);
+            assertFalse("NOR operator did not return false for a list of statements with a true and a false",
+                nor.getResult(context, null));
+            // Set to False, False (expect True)
+            nor.setStatements(falseStatements);
+            assertTrue("NOR operator did not return true for a list of false statements",
+                nor.getResult(context, null));
+        } catch (LogicalStatementException e) {
+            log.error(e.getMessage());
+            fail("LogicalStatementException thrown testing the NOR operator" + e.getMessage());
+        }
+    }
+
+    /**
+     * Test the NOT operator with simple individual true/false statements
+     */
+    @Test
+    public void testNotOperator() {
+        // Blank operator
+        Not not = new Not();
+        // Try tests - the item can be null, as the statements are simply returning booleans themselves
+        try {
+            // Set to True (expect False)
+            not.setStatements(trueStatementOne);
+            assertFalse("NOT operator did not return false for a true statement",
+                not.getResult(context, null));
+            // Set to False (expect True)
+            not.setStatements(falseStatementOne);
+            assertTrue("NOT operator did not return true for a false statement",
+                not.getResult(context, null));
+        } catch (LogicalStatementException e) {
+            log.error(e.getMessage());
+            fail("LogicalStatementException thrown testing the NOT operator" + e.getMessage());
+        }
+    }
+
+    /**
+     * Test a simple filter with a single logical statement: the MetadataValueMatchCondition
+     * looking for a dc.title field beginning with "TEST"
+     */
+    @Test
+    public void testMetadataValueMatchCondition() {
+        try {
+            MetadataValue metadataValueOne = metadataValueService.create(context, itemOne, metadataField);
+            MetadataValue metadataValueTwo = metadataValueService.create(context, itemTwo, metadataField);
+            metadataValueOne.setValue("TEST title should match the condition");
+            metadataValueTwo.setValue("This title should not match the condition");
```

Comment:
a test with an item without the checked metadata should be included

Reply:
That is what itemTwo is doing -- the check is for "title begins with TEST", and itemTwo has a title which is expected to not match.

Reply:
I'm asking to cover an additional use case when the logic filter involve a metadata that is not present in the item at all. This to proof that no NPE are raised in such condition or a false positive match

Reply:
Understood, I've added a further test for an item where the field is "dc.subject" which the item doesn't contain at all